mk_core/mk_event_kqueue.c

```diff
@@ -211,12 +212,19 @@ static inline int _mk_event_timeout_create(struct mk_event_ctx *ctx,
 #if defined(NOTE_SECONDS) && !defined(__APPLE__)
```

Comment:
I think this change is ok but I would invert the pre-processor conditional order to detect the availability of `NOTE_NSECONDS` before falling back to `NOTE_SECONDS`.

Also, could we fix the client code that exposed this bug as well so we have the safety net but don't actively rely on it?

Reply:
We actively rely on this part of macro condition on FreeBSD platform.

kqueue event system is only used for BSD like systems such as FreeBSD or macOS. Windows and Linux do not use this event system.

Reply:
Could you please clarify what you mean? I was trying to say that the conditional branch that detects `NOTE_NSECONDS` should preceed the one that detects `NOTE_SECONDS` because the latter would always be present and if left as is would take precedence and prevent the former from being taken.

I just raised that point because it happened a few days ago with a fluent-bit time related function.

Reply:
> Could you please clarify what you mean? I was trying to say that the conditional branch that detects `NOTE_NSECONDS` should preceed the one that detects `NOTE_SECONDS` because the latter would always be present and if left as is would take precedence and prevent the former from being taken.

Ah, I'd wrongly understood what you mean. Yes, we should detect `NOTE_NSECONDS` at first and detect `NOTE_SECONDS` the latter conditional branch. Thank you for clarifying this.